.github/workflows/ci.yaml

```diff
@@ -98,7 +104,35 @@ jobs:
           # Update whenever charmcraft.yaml is changed
           - series: jammy
             bases-index: 0
-    name: (GH hosted) ${{ matrix.groups.job_name }} | ${{ matrix.ubuntu-versions.series }}
+        juju-snap-channel:  ["2.9/stable", "3.1/candidate"]
+        include:
+          - juju-snap-channel: "3.1/candidate"
+            agent-version: "3.1.6"
+            libjuju-version:  "3.2.0.1"
+          - juju-snap-channel: "2.9/stable"
+            agent-version: "2.9.44.0"
+            libjuju-version:  "2.9.44.0"
+        exclude:
+          # Disabling HA tests, as long as we want to have a limited pipeline on Juju3
```

Comment:
Though I'm not @taurus-forever I'd have a small comment here :-)

I honestly think at least a minimal subset of Juju3 tests should be run on each PR.
The simple reason is (sry, I know I'm getting boring): *Juju Secrets*.

On a Juju2 pipeline NONE of the secrets-related workflows would be invoked. This means that no related bugs may be discovered, no related or theoretically-unrelated (modified) workflows should be run.

By now, if I'm correct, pretty much all DP charms execute at least Secrets related tests on Juju3, making sure all development considers both the old databag path, as much as Juju Secrets workflows.

Here I'm proposing to add a limited set of Juju3 tests on top of the full Juju2 workflows. 
Namely the basic tests together with `relation` tests. Without in-depth knowledge on MySQL Charm these seem to be likely to use internal secrets directly or indirectly furthermore `relations` cover external secrets usage as well (tests minimally extended in this PR)

Reply:
(Also in case you Guys may prefer to take more than the set above, I'd propose to merge this PR with the minimal set, and extend the list in a separate PR. Thus isolation of unrelated matters, and MySQL Charm is covered for External Secrets while further work on the pipelines)

Reply:
In my opinion, we cannot drop support for juju 2 as long as we have users who deploy with juju 2 (and I dont think that all users will migrate to juju 3 in the near future)

Reply:
With my limited understanging, agreeing to @shayancanonical I guess this is the best to do with these pipelines.

Have the "combined" pipeline for each PR, including a base, fairly light set of tests on areas where Juju3 features are known to be used (i.e. typically: Secrets :-) )

And then have a full pure Juju3 pipeline running every night.

Reply:
1) Juju 3 should have priority over Juju 2. Are we ready for this jump?
2) Juju 2.9 is LTS and our charm have to support it entire 2.9 lifetime (until 28.04) :-(

Reply:
All right, after chatting on this with @carlcsaposs-canonical : How about leaving the pipelines as they are for the sake of this PR, and modify (split) them later on?